org.eclipse.jdt.core/search/org/eclipse/jdt/internal/core/search/processing/JobManager.java

```diff
@@ -142,6 +168,7 @@ public void discardJobs(String jobFamily) {
 						}
 						currentJob.cancel();
 						it.remove();
+						notifyAll(); // notify waiters for awaitingJobsCount()
```

Comment:
Since we hold lock during the loop anyway (so waiters will stop sleeping but will be blocked), shouldn't this be called only once **after** the while loop?

Reply:
That should be no difference. Don't know if the compiler does that anyway.

Reply:
No, it doesn't (hint: use Bytecode view to see what compiler does). 
I would move that also below outside the loop. Just feels not natural to call that N times if in reality it only needs to be done once.

Reply:
It's the hotspot compiler which would do such optimizations at runtime. I don't care.